### PR TITLE
Adds LM Studio connector guide

### LINKED ISSUE #5474: Bring Your Own: Local Large Language Model (BYOL LLM)
### Description

What: We are introducing documentation around guidance for connecting a localy hosted LLM to Elastic with LM Studio. See documentation here: https://docs.google.com/document/d/10geJwx4UiXZILmP5eaeyuY5p_r6szpJhyp293uUQHIs/edit 

This should be placed under, as one of the options:

[Set up connectors for large language models (LLM)
](https://www.elastic.co/guide/en/security/current/llm-connector-guides.html) 

### Background & resources

* Point of contact: @Charelzard 

### Which documentation set does this change impact?

ESS and serverless

### ESS release

N/A

### Serverless release

N/A

### Feature differences

N/A

### API docs impact

N/A

### Prerequisites, privileges, feature flags

N/A

### PR DESCRIPTION
**Pull request by benironside**
Fixes #5474 by adding a guide for setting up an LLM connector to a local LLM using LM Studio. 

Preview: [Connect to your own local model using LM Studio](https://elastic-dot-co-docs-production-oceuyn7yw-elastic-dev.vercel.app/current/serverless/security/connect-to-byo-llm)

### CODE REVIEW

**Comment by elasticdocs** [comment]
:rocket: Built _elastic-dot-co-docs-preview-docs_ successfully!

* https://elastic-dot-co-docs-production-itb2a24k7-elastic-dev.vercel.app

* Built with commit 57c12066f393ba58dae0ba86a8c0a906b53a561b

> Issues? Visit #next-docs in Slack

File: docs/serverless/assistant/connect-to-byo-llm.mdx [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+<DocCallOut title="Important">
+Replace `<secret token>` with your actual token, and keep it safe since you'll need it to set up the ((elastic-sec)) connector. Replace `<yourdomainname.com>` with your actual domain name. If you decide to change the port number in LM studio to something other than 1234, update the `proxy_pass` value at the bottom of the configuration.
+</DocCallOut>
+
+### (Optional) set up performance monitoring for your reverse proxy
+You can use Elastic's [Nginx integration](https://www.elastic.co/docs/current/integrations/nginx) to monitor performance and populate monitoring dashboards in ((kib)).
+
+## Configure LM Studio and download a model
+
+First, install LM Studio. LM Studio supports the OpenAI SDK, which makes it compatible with Elastic's OpenAI connector, so you can connect Elastic AI Assistant to any model available in the LM Studio marketplace.
```
Comment (Charelzard) [inline docs/serverless/assistant/connect-to-byo-llm.mdx:87]: First, install [LM Studio](https://lmstudio.ai/). LM Studio supports the OpenAI SDK, which makes it compatible with Elastic's OpenAI connector, allowing you to connect any model available in the LM Studio marketplace.

File: docs/serverless/assistant/connect-to-byo-llm.mdx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+One current limitation of LM Studio is that when it is installed on a server, you must launch the application using its GUI before doing so using the CLI. For example, by using Chrome RDP with an [X Window System](https://cloud.google.com/architecture/chrome-desktop-remote-on-compute-engine). After you've opened the application the first time using the GUI, you can start it by using `sudo lms server start` in the CLI. 
+
+Once you've launched LM Studio, select a model: 
+
```
Comment (Charelzard) [inline docs/serverless/assistant/connect-to-byo-llm.mdx:92]: Insert updated screenshot, improved dimensions here.

Screenshot 2024-07-05 at 11 42 08 AM

**Review by Charelzard — COMMENTED** [review]
See final comments ready to publish!

File: docs/serverless/assistant/connect-to-byo-llm.mdx [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    1. Red for "Likely too large for this machine", which typically will not work.
+1. Download one or more models.
+
+In this example we used [`TheBloke/Mixtral-8x7B-Instruct-v0.1.Q3_K_M.gguf`](https://huggingface.co/TheBloke/Mixtral-8x7B-Instruct-v0.1-GGUF). It has 46.7B total parameters, a 32,000 token context window, and uses GGUF [quanitization](https://huggingface.co/docs/transformers/main/en/quantization/overview). For more information about model names and format information, refer to the following table.
+
+| Model Name | Parameter Size | Tokens/Context Window | Quantization Format |
+|------------|----------------|-----------------------|---------------------|
+| Name of model, sometimes with a version number. | LLMs are often compared by their number of parameters — higher numbers mean more powerful models. | Tokens are small chunks of input information. Characters do not necessarily correspond to tokens 1:1. You can use [Tokenizer](https://platform.openai.com/tokenizer) to see how many tokens a given prompt might contain. | Quantization reduces overall parameters and helps the model to run faster, but reduces accuracy. |
```
Comment (nastasha-solomon) [inline docs/serverless/assistant/connect-to-byo-llm.mdx:113]: Is there another, more accessible, way to write `1:1`? Maybe this:
 
```suggestion
| Name of model, sometimes with a version number. | LLMs are often compared by their number of parameters — higher numbers mean more powerful models. | Tokens are small chunks of input information. Characters do not necessarily correspond to tokens identically. You can use [Tokenizer](https://platform.openai.com/tokenizer) to see how many tokens a given prompt might contain. | Quantization reduces overall parameters and helps the model to run faster, but reduces accuracy. |
```

File: docs/serverless/assistant/connect-to-byo-llm.mdx [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+| Examples: Llama, Mistral, Phi-3, Falcon. | The number of parameters is a measure of the size and the complexity of the model. The more parameters a model has, the more data it can process, learn from, generate, and predict. | The context window defines how much information the model can process at once. If the number of input tokens exceeds this limit, input gets truncated. | Specific formats for quantization vary, most models now support GPU rather than CPU offloading. |
+
+
+## Load a model in LM Studio 
+
+After downloading a model, you'll need to load it in LM Studio using the GUI or LM Studio's [CLI tool](https://lmstudio.ai/blog/lms). You can do this either using the CLI, or the GUI.
```
Comment (nastasha-solomon) [inline docs/serverless/assistant/connect-to-byo-llm.mdx:119]: Minor edits to make this more direct.

```suggestion
After downloading a model, load it in LM Studio using the GUI or LM Studio's [CLI tool](https://lmstudio.ai/blog/lms). You can do this using the CLI or the GUI.
```

**Review by nastasha-solomon — COMMENTED** [review]
Left a few suggestions for your consideration. Overall, looks good!

File: docs/serverless/assistant/connect-to-byo-llm.mdx [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+After downloading a model, load it in LM Studio using the GUI or LM Studio's [CLI tool](https://lmstudio.ai/blog/lms). You can do this using the CLI or the GUI.
+
+### Option 1: load a model using the CLI (Recommended)
+
+It is a best practice to use the GUI to download at least one model from the marketplace first, and then load/unload the model(s) using the CLI. The GUI allows you to search for models, whereas the CLI only allows you to import specific paths, but the CLI provides a good interface for loading/unloading.
+
+Use the following commands in your CLI.
+
+1. Verify LM Studio: `lms`
```
Comment (natasha-moore-elastic) [inline docs/serverless/assistant/connect-to-byo-llm.mdx:128]:
```suggestion
1. Verify LM Studio is installed: `lms`
```

File: docs/serverless/assistant/connect-to-byo-llm.mdx [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+| Examples: Llama, Mistral, Phi-3, Falcon. | The number of parameters is a measure of the size and the complexity of the model. The more parameters a model has, the more data it can process, learn from, generate, and predict. | The context window defines how much information the model can process at once. If the number of input tokens exceeds this limit, input gets truncated. | Specific formats for quantization vary, most models now support GPU rather than CPU offloading. |
+
+
+## Load a model in LM Studio 
+
+After downloading a model, load it in LM Studio using the GUI or LM Studio's [CLI tool](https://lmstudio.ai/blog/lms). You can do this using the CLI or the GUI.
+
+### Option 1: load a model using the CLI (Recommended)
+
+It is a best practice to use the GUI to download at least one model from the marketplace first, and then load/unload the model(s) using the CLI. The GUI allows you to search for models, whereas the CLI only allows you to import specific paths, but the CLI provides a good interface for loading/unloading.
```
Comment (natasha-moore-elastic) [inline docs/serverless/assistant/connect-to-byo-llm.mdx:124]:
```suggestion
It is a best practice to download models from the marketplace using the GUI, and then load or unload them using the CLI. The GUI allows you to search for models, whereas the CLI only allows you to import specific paths, but the CLI provides a good interface for loading and unloading.
```

File: docs/serverless/assistant/connect-to-byo-llm.mdx [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+
+Use the following commands in your CLI.
```
Comment (natasha-moore-elastic) [inline docs/serverless/assistant/connect-to-byo-llm.mdx:126]:
```suggestion
Use the following commands in your CLI:
```

File: docs/serverless/assistant/connect-to-byo-llm.mdx [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+
+1. Verify LM Studio: `lms`
+2. Check LM Studio's status: `lms status`
+3. List all downloaded models: `lms ls`
+4. Load a model: `lms load`
+
+<DocImage url="images/lms-cli-welcome.png" alt="The CLI interface during execution of initial LM Studio commands"/>
```
Comment (natasha-moore-elastic) [inline docs/serverless/assistant/connect-to-byo-llm.mdx:133]: Should we blur out `charles_davison_elastic_co` ?

File: docs/serverless/assistant/connect-to-byo-llm.mdx [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+## Configure the connector in ((elastic-sec))
+
+Finally, configure the connector in ((kib)):
+
+1. Log in to ((kib)).
+2. Navigate to **Stack Management → Connectors → Create Connector → OpenAI**. The OpenAI connector enables this use-case because LM Studio uses the OpenAI SDK.
+3. Name your connector to help keep track of the model version you are using.
+4. Under **URL**, enter the domain name specified in your Nginx configuration file, followed by `/v1/chat/completions`.
+5. Under **Default model**, enter `local-model`.
+6. Under **API Key**, enter the secret token specified in your Nginx configuration file.
```
Comment (natasha-moore-elastic) [inline docs/serverless/assistant/connect-to-byo-llm.mdx:166]:
```suggestion
6. Under **API key**, enter the secret token specified in your Nginx configuration file.
```

File: docs/serverless/assistant/connect-to-byo-llm.mdx [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+7. Click **Save**.
+
+<DocImage url="images/lms-edit-connector.png" alt="The Edit connector page in ((kib)), with appropriate values populated"/>
+
+Setup is now complete. You can use the model you've loaded in LM Studio to power Elastic's generative AI features. You can test a variety of models as you interact with AI Assistant to see what works best without having to update your connector. 
+
+<DocCallOut title="Note">
+While local models work well for <DocLink slug="/serverless/security/ai-assistant" text="AI Assistant"/> if you would like to interact with <DocLink slug="/serverless/security/attack-discovery" text="Attack Discovery"/> we recommend you use one of <DocLink slug="/serverless/security/llm-performance-matrix" text="these models"/>. As local models become more performant over time, this is likely to change.
```
Comment (natasha-moore-elastic) [inline docs/serverless/assistant/connect-to-byo-llm.mdx:174]:
```suggestion
While local models work well for <DocLink slug="/serverless/security/ai-assistant" text="AI Assistant"/>, we recommend you use one of <DocLink slug="/serverless/security/llm-performance-matrix" text="these models"/>  for interacting with <DocLink slug="/serverless/security/attack-discovery" text="Attack Discovery"/>. As local models become more performant over time, this is likely to change. 
```

**Review by natasha-moore-elastic — COMMENTED** [review]
Left some minor suggestions, otherwise looks great!

**Review by natasha-moore-elastic — APPROVED** [review]
🚀

**Pull request merged**